src/Decimals.jl

```diff
@@ -3,35 +3,36 @@
 # @author jack@tinybike.net (Jack Peterson), 7/3/2014
 
 module Decimals
-    import Base: ==, +, -, *, /, <, float, inv, round, trunc
 
-    export Decimal,
-           decimal,
-           number,
-           normalize
+export Decimal,
+       decimal,
+       number,
+       normalize
 
-    const DIGITS = 20
+const DIGITS = 20
 
-    # Numerical value: (-1)^s * c * 10^q
-    struct Decimal <: AbstractFloat
-        s::Integer  # sign can be 0 (+) or 1 (-)
-        c::BigInt   # coefficient (significand), must be non-negative
-        q::Integer  # exponent
-    end
+# Numerical value: (-1)^s * c * 10^q
+struct Decimal <: AbstractFloat
+    s::Bool  # sign can be 0 (+) or 1 (-)
```

Comment:
Why don't we roll this into `c` ?

Reply:
I was actually wondering the same thing, I just didn't bother to change it.  I only spent about half an hour on this PR, mostly for the sake of type stability.  If the separate sign bit really is unnecessary, suggest somebody make another PR later to change it.

Reply:
Let me add another input. It appears that there is no standard for arbitrary precision decimal arithmetics (there is only [IEEE 754-2008](https://en.wikipedia.org/wiki/IEEE_754-2008_revision) standard introducing 32, 64, and 64-bit precisions - decimal32, decimal64 and decimal128, respectively, and these are already implemented in Julia package called [DecFP](https://github.com/JuliaMath/DecFP.jl)). Since `Decimals.jl` package aims at implementing arbitrary precision decimal arithmetics, it may _enjoy freedom_ from official standards. Good. Nonetheless, there is [General Decimal Arithmetic Specification by IBM](http://speleotrove.com/decimal/decarith.html), which extends the IEEE 754-2008. Perhaps it does not have the same status of official IEEE standards, but apparently, the now built-in Python package [decimal](https://docs.python.org/3.7/library/decimal.html) is implementing it. I tried to sketch these relationships at the bottom of [the front page (README.md)](https://github.com/JuliaMath/Decimals.jl).

Now, although I was not involved in the `Decimals.jl` package development (I only joined very recently) my perception of the package is that it might aim at the same goal as the Python's `decimal`. If this goal is shared, I think it would be wise to keep the [arithmetic model](http://speleotrove.com/decimal/damodel.html) as close as possible.

Reply:
I don't necessarily dispute any of that, but this PR was just a quick update mostly to get rid of the unnecessary type instability.  This package clearly could use a lot of work, I'd like it if we could merge this to get in what improvement we can for now and work can continue.

Reply:
We should move this discusion into #44